### PR TITLE
Payment Info Screen

### PR DESCRIPTION
**Pull request by Saagar3690**
### **Description**
Created Payment Info Screen with the following features:
1. Card Number is formatted automatically and only allows up the required 16 digits
2. Expiration Date automatically formatted to be MM/YY and requires all 4 digits
3. Basic verification for card number, expiry date, and CVV (i.e. they're all their required respective lengths, month for expiry date is between 1 and 12, inclusive, etc.
4. When the keyboard shows up, the content is all automatically shifted up, so it is not covered up by the keyboard
5. Ability to dismiss keyboard by touching any empty place on the screen
6. Navigates to PaymentSuccess screen that can pass a prop, success, of type boolean

Created Payment Success Screen that displays content based on the prop success that is passed with navigation 
- if success, displays, "Thank you for your purchase! Please take your item from the BruinBot."
- if success is false (i.e. useful for the future if payment processing with the Stripe API fails), displays "An error has occurred. Please try again." along with a button to navigate back to Payment Info Screen

### **Payment Screen In Action**
7233C95A-6611-4F5B-A630-D843FF5D
**If error has occurred in payment processing, it would display the screen below instead.**
IMG_5421

### **TODO**
Integrate Stripe API for payment processing

### CODE REVIEW

**Comment by dvorjackz** [comment]
Dayum that loading screen is super clutch @zhang-edward

File: src/containers/PaymentInfo.tsx
Comment (zhang-edward) [inline src/containers/PaymentInfo.tsx:82]: I just realized I wrote a `Form` component for this exact purpose of doing the keyboard avoiding and all that jazz. Maybe we would be able to use it here? But let's just add a todo for now

**Review by zhang-edward — DISMISSED** [review]
Added 1 comment but looks good!

File: src/containers/PaymentInfo.tsx
Comment (zhang-edward) [inline src/containers/PaymentInfo.tsx:112]: Could we use a `Button` component here (from react native elements) instead? 

also, let's disable the button while `loading` is true, since we don't want to charge cards multiple times.

**Review by zhang-edward — CHANGES_REQUESTED** [review]
just 1 very minor thing

File: src/containers/PaymentInfo.tsx
Comment (Saagar3690) [reply src/containers/PaymentInfo.tsx:82]: I don't see the Form component, so for now I've left a TODO to change KeyboardAvoidingView to use your form component

**Review by Saagar3690 — COMMENTED** [review]
### **Changes**
Updated TouchableOpacity to be Button from React-native-elements
Button is disabled after submission, but becomes enabled again if any of the three input fields are invalid
### **In Action**
6CCD6A6A-A560-4394-87F2-037D3D9A

### **TODO**
- Use Edward's form component instead of KeyboardAvoidingView
- Stripe API for payments processing
- if Stripe API processing fails, button needs to be enabled again when user presses go back from the Payment Success screen

**Comment by dvorjackz** [comment]
@zhang-edward seems like he addressed your comments so I will go ahead and approve this. @Saagar3690 you can def just leave a TODO for now, but have you checked out auth/form.tsx? Is that the one?

**Review by dvorjackz — APPROVED** [review]

**Comment by Saagar3690** [comment]
@dvorjackz @zhang-edward Yes I see the auth/form.tsx now since it was pushed recently. I'll pull and update the Payment to include the Form then push again

**Review by Saagar3690 — COMMENTED** [review]
### **Changes**
Replaced KeyboardAvoidingView and TouchableWithoutFeedback with Edward's Form component.

File: src/containers/PaymentInfo.tsx
Comment (dvorjackz) [inline src/containers/PaymentInfo.tsx:82]: Remove

**Review by dvorjackz — APPROVED** [review]
Looks good, feel free to remove that TODO, merge, and hit the squash and merge button

**Comment by Saagar3690** [comment]
@dvorjackz I don't have permissions to squash and merge since @zhang-edward hasn't approved it, so I have to wait for him to approve or you would have to merge in.

**Pull request merged**
